### PR TITLE
8310129: SetupNativeCompilation LIBS should match the order of the other parameters

### PR DESCRIPTION
**Pull request by TheShermanTanker**
I made the mistake of complicating the target pairs of LIBS and reversing the order they're specified in SetupNativeCompilation. They should match the sequence used in the other arguments, and since they're not used yet I'll take the opportunity to clean this up

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8310129](https://bugs.openjdk.org/browse/JDK-8310129): SetupNativeCompilation LIBS should match the order of the other parameters (**Bug** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14489/head:pull/14489` \
`$ git checkout pull/14489`

Update a local copy of the PR: \
`$ git checkout pull/14489` \
`$ git pull https://git.openjdk.org/jdk.git pull/14489/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14489`

View PR using the GUI difftool: \
`$ git pr show -t 14489`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14489.diff">https://git.openjdk.org/jdk/pull/14489.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14489#issuecomment-1592808478)

### CODE REVIEW

File: make/common/NativeCompilation.gmk [hunk 1]
```diff
@@ -1031,12 +1031,11 @@ define SetupNativeCompilationBody
...
   # Pickup extra OPENJDK_TARGET_OS_TYPE, OPENJDK_TARGET_OS and TOOLCHAIN_TYPE
   # dependent variables for LDFLAGS and LIBS, and additionally the pair dependent
-  # TOOLCHAIN_TYPE plus OPENJDK_TARGET_OS for LDFLAGS, or OPENJDK_TARGET_OS plus
-  # TOOLCHAIN_TYPE for LIBS
+  # TOOLCHAIN_TYPE plus OPENJDK_TARGET_OS
   $1_EXTRA_LDFLAGS += $$($1_LDFLAGS_$(OPENJDK_TARGET_OS_TYPE)) $$($1_LDFLAGS_$(OPENJDK_TARGET_OS)) \
       $$($1_LDFLAGS_$(TOOLCHAIN_TYPE)) $$($1_LDFLAGS_$(TOOLCHAIN_TYPE)_$(OPENJDK_TARGET_OS))
   $1_EXTRA_LIBS += $$($1_LIBS_$(OPENJDK_TARGET_OS_TYPE)) $$($1_LIBS_$(OPENJDK_TARGET_OS)) \
-      $$($1_LIBS_$(OPENJDK_TARGET_OS)_$(TOOLCHAIN_TYPE)) $$($1_LIBS_$(TOOLCHAIN_TYPE))
+      $$($1_LIBS_$(TOOLCHAIN_TYPE)_$(OPENJDK_TARGET_OS)) $$($1_LIBS_$(TOOLCHAIN_TYPE))
```
Comment (erikj79) [inline make/common/NativeCompilation.gmk:1038]: You should also change the order these are added together to match LDFLAGS. We need the most specific parameter to be added last. Later arguments override earlier arguments if there is a conflict between them.

**Comment by TheShermanTanker** [comment]
Is the new change what you meant? Sorry I don't think I caught on entirely

**Comment by TheShermanTanker** [comment]
I'll take that as a yes :P

Thanks!

**Pull request closed without merging**